sklearn/cluster/_hierarchical_fast.pyx

```diff
@@ -426,8 +426,9 @@ def single_linkage_label(L):
 
 # Implements MST-LINKAGE-CORE from https://arxiv.org/abs/1109.2378
 def mst_linkage_core(
-        const float64_t [:, ::1] raw_data,
-        DistanceMetric64 dist_metric):
+        const float64_t [:, ::1] data,
```

Comment:
All the modifications to this file seems unrelated to the issue this PR addresses. In case these are fixes, could you add them in a separate small PR?

Reply:
This, while being located in a different sub-package, is in fact related to the issue at hand (https://github.com/scikit-learn/scikit-learn/pull/28244#discussion_r1500954113)